token.go

```diff
@@ -0,0 +1,54 @@
+package cloudconfig
+
+import (
+	"fmt"
+	"time"
+
+	tokenoperator "github.com/giantswarm/kubeadm-token-operator/service/operator"
+	microerror "github.com/giantswarm/microkit/error"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/watch"
+)
+
+const (
+	WatchTimeout = 90 * time.Second
+)
+
+func SearchToken(k8sClient kubernetes.Interface, clusterID string) (string, error) {
```

Comment:
I think this is logic the operator should provide. For instance we have something similar in the operatorkit repo, but we agreed at some point to move such logic to the cert-operator in this specific case back then.